src/status_im/contexts/communities/overview/events.cljs

```diff
@@ -79,13 +83,19 @@
 (rf/reg-event-fx :communities/request-to-join request-to-join)
 
 (defn sign-data
-  [_ [community-id password sign-params]]
-  (let [addresses-to-reveal (map :account sign-params)]
-    {:fx [[:json-rpc/call
-           [{:method     "wakuext_signData"
-             :params     [(map #(assoc % :password password) sign-params)]
-             :on-success [:communities/request-to-join-with-signatures community-id addresses-to-reveal]
-             :on-error   [:communities/requested-to-join-error community-id]}]]]}))
+  [_ [{:keys [community-id password airdrop-address addresses-to-reveal]} sign-params]]
+  {:fx [[:json-rpc/call
+         [{:method     "wakuext_signData"
+           :params     [(map #(assoc % :password password) sign-params)]
+           :on-success [:communities/request-to-join-with-signatures
+                        {:community-id        community-id
+                         ;; NOTE: After completely migrating to the new reqest to join flow we can
```

Comment:
I read and re-read this comment, but I couldn't understand it. Could you elaborate a bit? Shouldn't we be using the feature flag to decide which path to take to make the events backwards compatible?

Reply:
My bad, I completely forgot that we can rely on the flag instead of using the passed value to make a decision. However, as I was running the tests, I realized the current approach of making a decision inside the event is a bad idea. The tests passed when the flag is off and failed when it is on. Relying on an outside value will make it hard to test. I think we should write new events to handle the new flow and keep the old ones untouched. What are your thoughts on this approach?